src/common/exception.cpp

```diff
@@ -175,6 +175,9 @@ const HTTPException &Exception::AsHTTPException() const {
 
 void Exception::ThrowAsTypeWithMessage(ExceptionType type, const string &message,
                                        const std::shared_ptr<Exception> &original) {
+	if (message == "std::bad_alloc") {
+		throw std::bad_alloc();
```

Comment:
Seems like it is impossible to wrap `std::bad_alloc` without triggering terminate, but not sure how to catch this here without a string-compare

Reply:
Maybe it's possible to do something with multiple inheritance, basically having something that's both a bad_alloc and an exception in our hierarchy?

I don't have a clear idea, something like what they describe here: https://stackoverflow.com/questions/22214128/custom-exception-hierarchy-dreaded-diamond-from-stdexception-and-stdbad-all
```c++
struct MyExceptionInfo { /* your extra functionality goes here */};

template<class T>
struct Ex : T, MyExceptionInfo
{
    template<class A1, class... Args>
    Ex(A1&& a1, Args&&... args)
        : T(std::forward<A1>(a1))
        , MyExceptionInfo(std::forward<Args>(args)...)
    {}
};

int main() {
    try {
        throw Ex<std::runtime_error>("oops");
    }
    catch(std::exception &e) {
        if(MyExceptionInfo* my_info = dynamic_cast<MyExceptionInfo*>(&e)) {
            // use MyExceptionInfo here
        }
    }
}
```
Unsure if this is any helpful